src/App.tsx

```diff
@@ -1,35 +1,47 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import styled from '@emotion/styled';
+import NavigationBar from '@/components/common/NavigationBar';
+import CategorySection from '@/components/home/CategorySection';
+import FriendSelectSection from '@/components/home/FriendSelectSection';
+import BannerSection from '@/components/home/BannerSection';
+import GiftRankingGrid from '@/components/GiftRanking/GiftRankingGrid';
+import LoginPage from '@/pages/loginpage';
+import NotFoundPage from '@/pages/NotFoundPage';
+import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
 
 function App() {
-  const [count, setCount] = useState(0)
-
   return (
-    <>
-      <div>
-        <a href="https://vite.dev" target="_blank">
-          <img src={viteLogo} className="logo" alt="Vite logo" />
-        </a>
-        <a href="https://react.dev" target="_blank">
-          <img src={reactLogo} className="logo react" alt="React logo" />
-        </a>
-      </div>
-      <h1>Vite + React</h1>
-      <div className="card">
-        <button onClick={() => setCount((count) => count + 1)}>
-          count is {count}
-        </button>
-        <p>
-          Edit <code>src/App.tsx</code> and save to test HMR
-        </p>
-      </div>
-      <p className="read-the-docs">
-        Click on the Vite and React logos to learn more
-      </p>
-    </>
-  )
+    <Router>
+      <AppContainer>
+        <AppInner>
+          <NavigationBar />
+          <Routes>
+            <Route path="/" element={
+              <>
+                <FriendSelectSection />
+                <CategorySection />
+                <BannerSection />
+                <GiftRankingGrid />
+              </>
+            } />
```

Comment:
페이지 수정 시 라우트 트리가 있는 파일을 수정하지 않아도 되도록 라우트와 1:1로 대응되는 페이지 컴포넌트를 만드는 것을 권장드립니다.

Reply:
이 내용에 대해서 정확하게 이해하지 못했는데 자세히 말씀해주실 수 있을까요...?

Reply:
메인 페이지 수정 시 App.tsx파일을 건드리지 않아도 되도록 페이지 컴포넌트를 분리해달라는 말이었어요!
```suggestion
            <Route path="/" element={MainPage} />
```